client/src/resources/checkers/AvailabilityChecker.js

```diff
@@ -1,6 +1,22 @@
 import { inject } from "aurelia-framework";
 import { Container } from "aurelia-dependency-injection";
 
+// This function transforms the existing check config to the new format
+// After all the tools adopted the new configuration format this is
+// not needed anymore
+function getConfig(availabilityCheck) {
+  const check = JSON.parse(JSON.stringify(availabilityCheck));
+  if (check.config) {
+    return check.config;
+  }
+  delete check.type;
```

Comment:
Can you add a deprecation notice here and also at all the other places where you check for the new vs. old way of configuration? Something like:
```suggestion
  log.info('DEPRECATION NOTICE: In Q editor 4.0 you will have to configure the availabilityCheck with a config property. See https://github.com/nzzdev/Q-editor/blob/master/README.md for details');
  delete check.type;
```

This also makes it clear, that this code branch can be removed for `4.0`.

Reply:
Added in https://github.com/nzzdev/Q-editor/pull/182/commits/ed35be9fec59e288d4dae456f9f1869527adc937